packages/web/server/api/edge-routers/concentrated-liquidity-router.ts

```diff
@@ -0,0 +1,297 @@
+import { CoinPretty, Dec, DecUtils, Int } from "@keplr-wallet/unit";
+import { tickToSqrtPrice } from "@osmosis-labs/math";
+import cachified, { CacheEntry } from "cachified";
+import { LRUCache } from "lru-cache";
+import { z } from "zod";
+
+import { DEFAULT_LRU_OPTIONS } from "~/config/cache";
+import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
+import { getAsset } from "~/server/queries/complex/assets";
+import { UserOsmoAddressSchema } from "~/server/queries/complex/parameter-types";
+import { queryPositionPerformance } from "~/server/queries/imperator";
+import { queryAddressCLPositions } from "~/server/queries/osmosis/concentrated";
+
+export const concentratedLiquidityRouter = createTRPCRouter({
+  getUserPositions: publicProcedure
+    .input(
+      z
+        .object({
+          sortDirection: z.enum(["asc", "desc"]).default("desc").optional(),
```

Comment:
I've been burned here where default doesn't actually get set by zod if an "optional" follows. Just a heads up (this may work)

Reply:
I've found that by adding `default`, we don't need to specify `optional`. The main change is that we have to use `z.input` instead of `z.infer` to determine the type. This makes sense because when used as input, a value can be optional, but when we're dealing with a type, we expect the value to always be defined.